enterprise/frontend/src/embedding-sdk/store/reducer.ts

```diff
@@ -113,26 +146,20 @@ export const sdk = createReducer(initialState, builder => {
     token: { ...state.token, loading: true },
   }));
 
-  builder.addCase(refreshTokenAsync.fulfilled, (state, action) => ({
-    ...state,
-    token: {
-      ...state.token,
+  builder.addCase(refreshTokenAsync.fulfilled, (state, action) => {
+    state.token = {
       token: action.payload,
-      error: null,
       loading: false,
-    },
-  }));
+      error: null,
+    };
 
-  builder.addCase(refreshTokenAsync.rejected, (state, action) => ({
-    ...state,
-    isLoggedIn: false,
-    token: {
-      ...state.token,
-      token: null,
-      error: action.error,
-      loading: false,
-    },
-  }));
+    state.loginStatus = { status: "success" };
```

Comment:
here and in the next reducer, i'm setting directly the loginStatus.
This avoids having to call setLoginStatus everywhere